src/modules/75-cf/pages/onboarding/__test__/OnboardingDetailPage.test.tsx

```diff
@@ -6,135 +7,91 @@
   test('OnboardingDetailPage empty state should be rendered properly', () => {
-    mockImport('@cf/hooks/useEnvironmentSelectV2', {
-      useEnvironmentSelectV2: () => ({
-        loading: true,
-        refetch: jest.fn(),
-        EnvironmentSelect: <div />,
-        environments: [
-          {
-            accountId: 'harness',
-            identifier: 'foo',
-            name: 'bar',
-            type: 'Production'
-          }
-        ]
-      })
-    })
-
-    const { container } = render(
-      <TestWrapper
-        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/onboarding/detail"
-        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
-      >
-        <OnboardingDetailPage />
-      </TestWrapper>
-    )
-    expect(container).toMatchSnapshot()
+    renderComponent()
+
+    // Breadcrumbs
+    expect(screen.getByTestId('getStartedBreadcrumb')).toHaveTextContent('cf.shared.getStarted/cf.shared.quickGuide/')
+    // First tab should be selected
+    expect(screen.getByText('cf.onboarding.oneCreateAFlag')).toBeVisible()
+    expect(screen.getByText('cf.onboarding.oneCreateAFlag').parentElement).toHaveAttribute('aria-selected', 'true')
+
+    // Footer buttons
+    expect(screen.getByText('next')).toBeVisible()
+    expect(screen.getByText('next').closest('button')).toBeDisabled()
+    expect(screen.getByText('back')).toBeVisible()
   })
 
   test('Should be able to create a flag', () => {
-    mockImport('@cf/hooks/useEnvironmentSelectV2', {
-      useEnvironmentSelectV2: () => ({
-        loading: true,
-        refetch: jest.fn(),
-        EnvironmentSelect: <div />,
-        environments: [
-          {
-            accountId: 'harness',
-            identifier: 'foo',
-            name: 'bar',
-            type: 'Production'
-          }
-        ]
-      })
-    })
-
-    const { container } = render(
-      <TestWrapper
-        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/onboarding/detail"
-        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
-      >
-        <OnboardingDetailPage />
-      </TestWrapper>
-    )
-
+    const createNewFlag = jest.spyOn(ffServices, 'useCreateFeatureFlag')
+    const setSelectedFlag = jest.fn()
     const flagName = 'test-boolean-flag'
 
-    fireEvent.input(container.querySelector('input') as HTMLInputElement, { target: { value: flagName } })
-
-    waitFor(() =>
-      expect(container.querySelector('button[class*=Button--button]:not(:first-of-type)[class*=disabled]')).toBeNull()
-    )
+    renderComponent()
 
-    fireEvent.click(container.querySelector('button[class*=Button--button]:not(:first-of-type)') as HTMLButtonElement)
+    // Should be prevented from next tab until flag selected/created
+    expect(screen.getByText('next')).toBeVisible()
+    expect(screen.getByText('next').closest('button')).toBeDisabled()
```

Comment:
might be slightly better to do:
```suggestion
    expect(screen.getByRole('button', {name: 'next'})).toBeDisabled()
```

Reply:
I'll try this again, it wasn't able to find any buttons this way for some reason when I tried this in `SelectEnvironmentView`

Reply:
Doesn't work here without making changes to the html itself, will look at improving this when back working in this area